### PR TITLE
Update HPKE info/AAD strings

### PR DESCRIPTION
**Pull request by cjpatton**
Partially addresses #8.
Based on #17 (merge that first).

Aligns the info and AAD strings with draft-ietf-ppm-dap-01.

### CODE REVIEW

File: daphne/src/vdaf/mod.rs [hunk 1]
```diff
@@ -196,17 +195,18 @@ impl VdafConfig {
...
+        const N: usize = CTX_INPUT_SHARE.len();
+        let mut info = [0; N + 2];
+        info[..N].copy_from_slice(CTX_INPUT_SHARE);
+        info[N] = CTX_ROLE_CLIENT; // Sender role
+        info[N + 1] = if is_leader {
             CTX_ROLE_LEADER
         } else {
             CTX_ROLE_HELPER
-        });
+        }; // Redeiver role
```
Comment (armfazh) [inline daphne/src/vdaf/mod.rs:206]:
```suggestion
        }; // Receiver role
```

File: daphne/src/vdaf/mod.rs [hunk 2]
```diff
@@ -196,17 +195,18 @@ impl VdafConfig {
...
 
-        let mut aad = Vec::with_capacity(10);
+        let mut aad = Vec::with_capacity(50 + extensions.len());
+        task_id.encode(&mut aad);
         nonce.encode(&mut aad);
```
Comment (armfazh) [inline daphne/src/vdaf/mod.rs:210]: is it ok the calculation of the nonce length? now task_id was prepended.

File: daphne/src/vdaf/mod.rs [hunk 3]
```diff
@@ -697,20 +697,18 @@ impl VdafConfig {
...
-        info.extend_from_slice(CTX_AGG_SHARE_PREFIX);
-        info.push(0); // Sender role (overwritten below)
-        let sender_role_index = info.len() - 1;
-        info.push(CTX_ROLE_COLLECTOR);
-
-        let mut aad = Vec::with_capacity(8);
+        const N: usize = CTX_AGG_SHARE.len();
+        let mut info = [0; N + 2];
+        info[..N].copy_from_slice(CTX_AGG_SHARE);
+        info[N + 1] = CTX_ROLE_COLLECTOR; // Reciver role (sender role set below)
```
Comment (armfazh) [inline daphne/src/vdaf/mod.rs:703]:
```suggestion
        info[N + 1] = CTX_ROLE_COLLECTOR; // Receiver role (sender role set below)
```

**Review by armfazh — APPROVED** [review]

File: daphne/src/vdaf/mod.rs (hunk 2 above)
Comment (cjpatton) [reply daphne/src/vdaf/mod.rs:210]: Nope, it's wrong!! I discovered that when confirming interop with Janus. Will fix in this PR before merging.

Comment (cjpatton) [reply daphne/src/vdaf/mod.rs:211]: Fixed. Great catch!

**Pull request merged**